### PR TITLE
feat: moved action to use GitHub Package Registry

### PR DESCRIPTION
**Pull request by Bullrich**
Updated the project to store an already built image in the GitHub Package Registry.

This PR brings 3 changes:
- It creates a new release CI flow.
- It modifies the Docker image to be a multi step image.
- It changes the action definition file (`action.yml`) to use an image from the registry.

## The CI

The new CI workflow does the following:
- Checks that the Docker image can be built.
- Checks that the version for the package and the docker image are the same
- If the version in the `package.json` changed:
  - Tag the commit with the new version
  - Generate a new release (with a changelog)
  - Builds and publishes the docker image using the new tag

## The Dockerimage

As the image is being stored, I modified it to actually use a second slim version with the compiled code so it is smaller (and gets downloaded faster).

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          fetch-depth: 0
+      - uses: butlerlogic/action-autotag@stable
+        id: autotag
+        with:
+          head_branch: master
+          tag_prefix: "v"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Changelog
+        uses: Bullrich/generate-release-changelog@master
```
Comment (rzadp) [inline .github/workflows/publish.yml:66]: Can we pin a version or a commit hash?

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,19 @@
-FROM node:18
+FROM node:18 as Builder
+
+WORKDIR /action
+
+RUN npm i -g @vercel/ncc
```
Comment (rzadp) [inline Dockerfile:5]: Can we pin the version here?

File: action.yml [hunk 3]
```diff
@@ -29,4 +29,4 @@ inputs:
     type: string
 runs:
   using: 'docker'
-  image: 'Dockerfile'
+  image: 'docker://ghcr.io/bullrich/github-issue-sync/example:0.3.1'
```
Comment (rzadp) [inline action.yml:32]:
```suggestion
  image: 'docker://ghcr.io/paritytech/github-issue-sync/example:0.3.1'
```

Right?

File: Dockerfile (hunk 2 above)
Comment (Bullrich) [reply Dockerfile:5]: Let's add it to the build dev dependencies instead, so dependabot can keep it updated. What do you think?

File: action.yml (hunk 3 above)
Comment (Bullrich) [reply action.yml:32]: OMG, great catch! I forgot about that!

File: Dockerfile (hunk 2 above)
Comment (rzadp) [reply Dockerfile:5]: Perfect.

**Review by mutantcornholio — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+env:
+  IMAGE_NAME: action
+
+jobs:
+  test-image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3.3.0
+      - name: Run tests
```
Comment (rzadp) [inline .github/workflows/publish.yml:16]:
```suggestion
      - name: Check that the image builds
```

File: Dockerfile [hunk 5]
```diff
@@ -1,4 +1,8 @@
...
```
Comment (rzadp) [inline Dockerfile:5]: Should this line be removed now that it's in `yarn build`?

Comment (Bullrich) [reply Dockerfile:5]: Oh I thought that I removed it. Doing it now

**Review by rzadp — APPROVED** [review]

**Pull request merged**